vck-openid/src/commonMain/kotlin/at/asitplus/wallet/lib/oidvci/Extensions.kt

```diff
@@ -208,32 +214,101 @@ suspend fun Issuer.IssuedCredential.toCredentialResponseSingleCredential(
 private fun Issuer.IssuedCredential.Iso.toBase64UrlStrict(): String =
     issuerSigned.serialize().encodeToString(Base64UrlStrict)
 
-sealed class OAuth2Exception : Throwable {
-    constructor(error: String, errorDescription: String, cause: Throwable?) : super("$error: $errorDescription", cause)
-    constructor(error: String, errorDescription: String) : super("$error: $errorDescription")
-    constructor(error: String, cause: Throwable?) : super(error, cause)
-
-    class InvalidRequest(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_REQUEST, message, cause)
-    class InvalidClient(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_CLIENT, message, cause)
-    class InvalidScope(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_SCOPE, message, cause)
-    class InvalidGrant(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_GRANT, message, cause)
-    class InvalidCode(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_CODE, message, cause)
-    class InvalidToken(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_TOKEN, message, cause)
-    class InvalidProof(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_PROOF, message, cause)
-    class UserCancelled(message: String, cause: Throwable? = null) : OAuth2Exception(USER_CANCELLED, message, cause)
-    class InvalidDpopProof(message: String, cause: Throwable? = null) :
-        OAuth2Exception(INVALID_DPOP_PROOF, message, cause)
-
-    class UnsupportedCredentialType(message: String, cause: Throwable? = null) :
-        OAuth2Exception(UNSUPPORTED_CREDENTIAL_TYPE, message, cause)
-
-    class CredentialRequestDenied(message: String, cause: Throwable? = null) :
-        OAuth2Exception(CREDENTIAL_REQUEST_DENIED, message, cause)
-
-    class InvalidNonce(message: String, cause: Throwable? = null) : OAuth2Exception(INVALID_NONCE, message, cause)
-    class AccessDenied(message: String, cause: Throwable? = null) : OAuth2Exception(ACCESS_DENIED, message, cause)
-    class RegistrationValueNotSupported(message: String, cause: Throwable? = null) :
-        OAuth2Exception(REGISTRATION_VALUE_NOT_SUPPORTED, message, cause)
-}
+@Serializable
+sealed class OAuth2Exception(
```

Comment:
Please check with @md4096 if this matches his implementation of transporting errors to verifiers